core/src/com/rallyhealth/weepickle/v1/core/Visitor.scala

```diff
@@ -284,6 +285,33 @@ object Visitor {
 
     def visitEnd(): Z = f(src.visitEnd())
   }
+
+  /*
+   * Delegates everything verbatim, except visitObject.visitKey maps the key.
+   * Good for things like going to/from snake case.
+   */
+  class MapKeys[T, J](delegatedTo: Visitor[T, J], keyMapper: CharSequence => CharSequence) extends Delegate[T, J](delegatedTo) {
+
+    override def visitObject(length: Int): ObjVisitor[T, J] =
+      new MapKeyObjDelegate[T, J](delegatedTo.visitObject(length), keyMapper)
+
+    override def visitArray(length: Int): ArrVisitor[T, J] =
+      new MapKeyArrDelegate[T, J](delegatedTo.visitArray(length), keyMapper)
+  }
+
+  class MapKeyObjDelegate[T, J](objVisitor: ObjVisitor[T, J], keyMapper: CharSequence => CharSequence) extends ObjDelegate[T, J](objVisitor) {
```

Comment:
Can we make "Key" plural for consistency with MapKeys?

Reply:
done